### PR TITLE
feat: Helm template secrets

### PR DESCRIPTION
**Pull request by matteocellucci**
### Type of changes

- [x] Add new resources
- [ ] Update configuration to existing resources
- [ ] Remove existing resources

### Env to apply

- [x] DEV
- [ ] UAT
- [ ] PROD

### Does this introduce a change to production resources with possible user impact?

- [x] Yes, users may be impacted applying this change
- [ ] No

### Does this introduce an unwanted change on infrastructure? Check terraform plan execution result

- [ ] Yes
- [x] No

### Other information

---

### CODE REVIEW

File: src/core/03_aks.tf [hunk 1]
```diff
@@ -29,7 +29,7 @@ module "k8s_snet" {
 }
 
 module "aks" {
-  source = "git::https://github.com/pagopa/azurerm.git//kubernetes_cluster?ref=2.3.1"
+  source = "git::https://github.com/pagopa/azurerm.git//kubernetes_cluster?ref=kv-secrets-autorotation"
```
Comment (matteocellucci) [inline src/core/03_aks.tf:32]: To change after merge of https://github.com/pagopa/azurerm/pull/196

Comment (pasqualedevita) [reply src/core/03_aks.tf:32]:
```suggestion
  source = "git::https://github.com/pagopa/azurerm.git//kubernetes_cluster?ref=v2.4.0"
```

**Review by diegolagospagopa — APPROVED** [review]

File: src/k8s/04_csi_key_vault_access_policy.tf [hunk 2]
```diff
@@ -0,0 +1,7 @@
+resource "azurerm_key_vault_access_policy" "cluster_access_policy" {
+  key_vault_id = data.azurerm_key_vault.kv.id
+  tenant_id    = data.azurerm_subscription.current.tenant_id
+  object_id    = "25859118-4d19-418a-97ff-97af3a4ca929"
```
Comment (pasqualedevita) [inline src/k8s/04_csi_key_vault_access_policy.tf:4]: can we make this value general? if it's complex get managed identity we can create a custom user assigned managed identity

**Review by pasqualedevita — CHANGES_REQUESTED** [review]

**Review by pasqualedevita — APPROVED** [review]
awesome!! 🥇

**Pull request merged**